### PR TITLE
DSD-240: MultiSelect Draft

### PR DESCRIPTION
**Pull request by nypl-wluisi**
Fixes JIRA ticket xxx

## This PR does the following:
-

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] All new and existing tests passed.

### Front End Review:

- [x] View [the example in Storybook](https://pr776-j6te1xet1hg8tkltm8lz7twsjidrkxrl.tugboat.qa/)

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr776-j6te1xet1hg8tkltm8lz7twsjidrkxrl.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6197d859e368650741427363

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr776-j6te1xet1hg8tkltm8lz7twsjidrkxrl.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6197d859e368650741427363

**Comment by nypl-wluisi** [comment]
@bigfishdesign13 I at least got this branch updated with the latest from development, as it was way behind, so tugboat preview should work again.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr776-j6te1xet1hg8tkltm8lz7twsjidrkxrl.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6197d859e368650741427363

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr776-j6te1xet1hg8tkltm8lz7twsjidrkxrl.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6197d859e368650741427363

File: src/components/FilterBar/FilterBar.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  /** The label of the FilterBar. */
+  label: string;
+  /** Whether the FilterBar modal is open or closed. */
+  isModalOpen: boolean;
+  /** The action to perform on the "filters" button onClick function. */
+  onModalToggle: () => void;
+  /** Whether the FilterBar is in mobile mode or not. */
+  isMobile: boolean;
+  /** The selected items (items that were checked by user). */
+  selectedItems?: SelectedItems;
```
Comment (bigfishdesign13) [inline src/components/FilterBar/FilterBar.tsx:24]: @nypl-wluisi With the `MultiSelect` components coming in as children, is the consuming app expected to maintain the state of the `MultiSelect` components in the `Filterbar`?  I see that this prop is used to generate the label text for the mobile "show filters” button.  Is this value also returned in the `onApply` callback?

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr776-gpuszll0afk8knlvlljt0xl0zagjmupm.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/62394810c49d661fdca706ec

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr776-jwrdqr2ttir6ylbgyzwrybmlbqzk7qw0.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/623e281897feaa22ee12e9b5

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr776-jwrdqr2ttir6ylbgyzwrybmlbqzk7qw0.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/623e281897feaa22ee12e9b5

**Comment by nypl-wluisi** [comment]
Closing this in favor of https://github.com/NYPL/nypl-design-system/pull/987

**Pull request closed without merging**